lib/chef/provider/remote_file/http.rb

```diff
@@ -140,6 +140,10 @@ def http_client_opts
           opts
         end
 
+        def http_options
+          http_client_opts.merge(new_resource.http_options)
+        end
```

Comment:
this is simple enough it should probably be merged into the existing `http_client_opts` method

Reply:
Updated, Please have a look.